File: Talaqi-Backend/Talaqi.Soluation/src/Core/Talaqi.Domain/Entities/Reporting/Report.cs
using System;
using Talaqi.Domain.Enums;

namespace Talaqi.Domain.Entities.Reporting
{
    public class Report : BaseEntity
    {
        public Guid ReporterId { get; set; }
        public virtual User Reporter { get; set; } = null!;

        public ReportTargetType TargetType { get; set; }

        // Context References
        public Guid? TargetUserId { get; set; }
        public virtual User? TargetUser { get; set; }

        public Guid? ConversationId { get; set; }
        public virtual Talaqi.Domain.Entities.Messaging.Conversation? Conversation { get; set; }

        public Guid? MessageId { get; set; }
        public virtual Talaqi.Domain.Entities.Messaging.Message? Message { get; set; }

        public ReportReason Reason { get; set; }
        public string? Description { get; set; }

        public ReportStatus Status { get; set; } = ReportStatus.Pending;

        // Admin Actions
        public string? AdminNotes { get; set; }
        public Guid? ResolvedById { get; set; }
        public virtual User? ResolvedBy { get; set; }
        public DateTime? ResolvedAt { get; set; }
    }
}

File: Work-Updates/Talaqi-Frontend/talaqi-frontend/src/app/core/models/report.ts
export enum ReportStatus {
  Pending = 0,
  UnderReview = 1,
  Resolved = 2,
  Rejected = 3
}

export enum ReportTargetType {
  User = 0,
  Conversation = 1,
  Message = 2,
  General = 3
}

export enum ReportReason {
  Spam = 0,
  Harassment = 1,
  InappropriateContent = 2,
  FakeItem = 3,
  Scam = 4,
  Other = 5
}

export interface CreateReportDto {
  targetType: ReportTargetType;
  targetUserId?: string;
  conversationId?: string;
  messageId?: string;
  reason: ReportReason;
  description?: string;
}

export interface ReportDto {
  id: string;
  reporterId: string;
  reporterName: string;
  targetType: ReportTargetType;
  targetUserId?: string;
  targetUserName?: string;
  conversationId?: string;
  messageId?: string;
  reason: ReportReason;
  description?: string;
  status: ReportStatus;
  adminNotes?: string;
  createdAt: string;
  updatedAt?: string;
}

export interface UpdateReportStatusDto {
  status: ReportStatus;
  adminNotes?: string;
}

export interface ReportFilterDto {
  status?: ReportStatus;
  reason?: ReportReason;
  targetType?: ReportTargetType;
  fromDate?: string;
  toDate?: string;
  searchTerm?: string;
  page: number;
  pageSize: number;
}
